gog/graph.go

```diff
@@ -47,6 +47,16 @@ type Grapher struct {
 
 	seenDocObjs map[types.Object]struct{}
 	seenDocKeys map[string]struct{}
+
+	// pkgDefs accumulates the defs from the package being graphed currently in Graph().
+	// If the package is graphed successfully, the defs are added to Output.
+	pkgDefs []*Def
```

Comment:
But `gog` continues to operate at the source unit level. So if you want the old behavior (failing an entire source unit for any error in def), you can just rely on https://github.com/sourcegraph/srclib-go/pull/82/files/765513444c12e5fbc848d13d2e5c5d8e87457a69#diff-a16471055e07a5dfd4d03aec2d1e38fbL371.

It's the `Graph` function in `graph.go` that handles multiple source units and has to be robust to failures within a given source unit (which you accomplish by changing the error returns to `log.Errorf` calls).